mappings/dnd5e.mapping

```diff
@@ -93,30 +93,38 @@
+    { "pdf": "Wpn3 Damage", "foundry": (function() { const dda = Array.from(actor.itemTypes.weapon.filter(i => i.data.data.equipped && i.hasAttack && i.hasDamage))?.[2]?.labels.derivedDamage; return !dda ? "" : dda.map(dd => `${dd.formula} ${game.dnd5e.config.damageTypes[dd.damageType]}`).join('\n')})()},
     { "pdf": "SleightofHand", "foundry": @data.skills.slt.total },
     { "pdf": "CHamod", "foundry": @data.abilities.cha.mod },
     { "pdf": "Survival", "foundry": @data.skills.sur.total },
     { "pdf": "AttacksSpellcasting", "foundry": "" },
     { "pdf": "Passive", "foundry": @data.skills.prc.passive },
-    { "pdf": "CP", "foundry": @data.currency.cp },
-    { "pdf": "ProficienciesLang", "foundry":
-        [
-            actor.data.data.traits.languages.value.filter(String).map(l => l.capitalize()).join(", "),
-            actor.data.data.traits.languages.custom,
-            actor.data.data.traits.weaponProf.value.filter(String).map(l => l.capitalize()).join(", "),
-            actor.data.data.traits.weaponProf.custom,
-            actor.data.data.traits.toolProf.value.filter(String).map(l => l.capitalize()).join(", "),
-            actor.data.data.traits.toolProf.custom,
-            actor.data.data.traits.armorProf.value.filter(String).map(l => l.capitalize()).join(", "),
-            actor.data.data.traits.armorProf.custom,
-        ].filter(String).join(", ")
+    { "pdf": "CP", "foundry": @data.currency.cp || "" },
+    { "pdf": "ProficienciesLang", "foundry": (function() {
+            let s = "";
+            let a = @data.traits.weaponProf.value.map(x => game.dnd5e.config.weaponProficiencies[x]
+                || game.packs.get("dnd5e.items").index.get(game.dnd5e.config.weaponIds[x])?.name)
+                .concat(@data.traits.weaponProf.custom.split(";")).filter(x => String(x) && x.length);
+            if (a.length > 0) { s = `${s}Weapons: ${a.join(", ")}\n` }
+            a = @data.traits.armorProf.value.map(x => game.dnd5e.config.armorProficiencies[x]
+                || game.packs.get("dnd5e.items").index.get(game.dnd5e.config.armorIds[x])?.name)
+                .concat(@data.traits.armorProf.custom.split(";")).filter(x => String(x) && x.length);
+            if (a.length > 0) { s = `${s}Armor: ${a.join(", ")}\n` }
+            a = @data.traits.toolProf.value.map(x => game.dnd5e.config.toolProficiencies[x]
+                || game.packs.get("dnd5e.items").index.get(game.dnd5e.config.toolIds[x])?.name)
+                .concat(@data.traits.toolProf.custom.split(";")).filter(x => String(x) && x.length);
+            if (a.length > 0) { s = `${s}Tools: ${a.join(", ")}\n` }
+            a = @data.traits.languages.value.map(x => game.dnd5e.config.languages[x])
+                .concat(@data.traits.languages.custom.split(";")).filter(x => String(x) && x.length);
+            if (a.length > 0) { s = `${s}Languages: ${a.join(", ")}\n` }
+            return s
```

Comment:
This is super minor and unimportant, but I suggest putting a semicolon at the end of the line here.